swagger.yaml

```diff
@@ -4294,7 +4294,7 @@ definitions:
         type: string
         description: "The desired network type used."
         enum: ['OpenShiftSDN', 'OVNKubernetes']
-        default: 'OpenShiftSDN'
+        x-nullable: true
```

Comment:
Not sure how I feel about this. This kinda feels like a breaking change - previously the default value was explicitly `OpenShiftSDN` and users could rely on it always being `OpenShiftSDN` when they omit this parameter. Suddenly it can take an entirely different value.

But actually we've already broke that contract once we've started forcing OVNKubernetes on Single Node installations, so it's not really this PR's fault, this PR is just being explicit about us breaking it instead of having the swagger lie...

Reply:
I only added changes to internal/controller/controllers/clusterdeployments_controller.go and internal/controller/controllers/clusterdeployments_controller_test.go, sorry for that